### ISSUE TITLE
Add missions to the extended tutorial system

### ISSUE THREAD

**Issue by gilbertohasnofb** [Juanvvc/c172p-detailed#52]
TUT02 - Add missions to the extended tutorial system

**Comment by Juanvvc** [comment]
Since we have currently plenty of time to the next release, I would like to develop this issue. My idea is providing several missions covering some of the special features of our aircraft. These missions could also be used as tutorials for the newcomers.

In addition, we can use these missions as a kind of "personal signature" for our team. I propose we all to chose a location for these missions in each of our countries. Also, please, consider some area with gilberto's new regional textures! Don't worry if the scenery is not yet developed, we have plenty of time. We can add a few buildings and submit them to terrasync.

Some ideas:

- Take off and Landing tutorial: we can develop a "precision landing competition": land as close as possible to a mark with several configurations: normal, no-engine, no-flaps. I attended this kind of competitions as a pilot and are very fun! Check this thread: http://forum.flightgear.org/viewtopic.php?f=79&t=22831 This mission was nearly completed, but I must check if I still have the files. I propose LECD, Spain. A small airport in the middle of the Pyrenees, just on the French-Spanish border. The airport has already some buildings made by me :)
- Crash system: starting in mid-air (if possible), the engine fails (for example, the fuel is exhausted) Your goal is locating and landing on a nearby airport without crashing.
- Navigate between two close airports using VFR or IFR. I think a flight of 30 minutes or one hour is the best idea for this mission.
- Navigate between two lakes. I guess the great lakes in USA or Alaska are the best sceneries for this mission. I prefer if there is some seaport available!

**Comment by gilbertohasnofb** [comment]
@Juanvvc I love this idea! We could also add:

- Amphibian: land the plane using the amphibian configuration in a nearby lake (we can chose a place which is not very large so that the approach has to be a bit more precise, as opposed to landing in the ocean where you never run out of "runway" :wink: )

We could also add some other missions which are not directly connected with special features of our plane, but which some people (particularly non-hardcore simmers) would appreciate, and which would require very little addition to the scenery. Some examples:

- if we add a static plane to some mountains, we can make a mission in which the pilot has to help find a plane that went missing. 
- a mission in which the pilot has to deliver goods to a remote location (maybe using the bush kit?)
- a time-limited mission in which the pilot has to bring medical personal to another remote location

Also, I just checked LECD and it is looking really nice (is that swimming pool by the runway really there in real life? That sounds like fun!), so we can use it for one of these missions. We could also make one in Kansas so people using @wlbragg 's Kansas custom scenery can enjoy it.

**Comment by tigert** [comment]
Diversion: flight planned and progressing on a vor radial as a starting point for example. But bad weather front lies ahead and you need to divert to an alternate field. X-plane for ipad has nice tutorials (greetings from Crete ;)) and one of them is such diversion, where one extra thing to consider is the main runway being out of service, thus needing to figure out, identify and land on a different runway.

**Comment by Juanvvc** [comment]
Another idea: formation flying. Follow a ghost aircraft using some max and min vertical and horizontal separation distance.

**Comment by wlbragg** [comment]
Funny, on my to do list is a huge project I have already the rough outline for and that is a "Major" Mission Add-on.

**Comment by wlbragg** [comment]
One thing on the top of my list was a night time (and daytime) search for the bad guy or lost hiker.

Or, in this part of the world a common mission for small aircraft is Marijuana eradication, find the grove.

Life flight was another, (timed). With the Cessna though it might have to be from hospital to hospital (city to city).

**Comment by gilbertohasnofb** [comment]
> It was something I really enjoyed with the "original" MS Flight Sim. 

What I loved when I first bought FS2002 was the training with the c172p. It was something like this: Lesson 1 flying straight and level (so if you drop too much altitude or if the vertical speed is too high you fail), Lesson 2 coordinate turns while flying straight (turns of no more than 30 degrees), etc. 

And all of this with some voice explaining it (the aircraft was always set to autopilot while the voice explained, so that you felt as if the instructor was there in control at that time).

Quite neat feature which I loved when I was young :smile:

**Comment by gilbertohasnofb** [comment]
@Juanvvc 
> yes, the swimming pool is between the runway and the tower/restaurant :)

Spending the day in the swimming pool watching planes, sounds like my kind of fun :smile:

**Comment by wkitty42** [comment]
PROPOSAL: amphibious take off and landing... take off from KRDU and fly the few miles to jordan lake for an amphibious landing near the US64 bridge that crosses the lake... then beach the craft on the western side of the lake near the bridge and go camping for a take off and return to KRDU later...

**Comment by Juanvvc** [comment]
I'm preparing two new missions, as well as cleaning up the tutorials:

- Precision landings. Not sure if at LEAT (I participated in a precision landing competition over there) or at LECD (much nicer scenery) Expected challenge: get the best score! This mission may work as a "landing tutorial".
- Floats. Fly from two lakes in Como (Italy) to Lachen (Switzerland) across the Alps. Challenge: did I say "across the Alps"? The direct route is not possible on our little Cessna. Find your way and keep in eye on the fuel level! This mission may work as a "VFR cross-country" because the pilot must plan the route before starting the simulator.
- Skis. Fly from ??? to ???. Challenge: bad weather? I'm thinking about a short fly to transport clients between an airport and some isolated lodge. It is similar to wkitty42's proposal, and to the service these guys provide near Anchorage (Alaska): http://www.tordrillomountainlodge.com/ I'd need the help of our collaborators in the USA to define this mission and its best location ;)
- I really, really want a mission in South America to honor the wonderful gilberto's textures in our aircraft and terrain. Do you have any ideas for this? Tigert's idea about diversions seems funny. Controlling the weather during a mission is not straightforward, but I'll think about it.

Currently, I'm really stuck on creating dialogs in canvas to show a list of available missions and relevant information about them. I'm afraid the canvas GUI is still extremely limited. I'm planning to create a branch with all this during this weekend.

**Comment by wlbragg** [comment]
Really cool, for the skis/or floats mission I can suggest a few locations in Alaska. There are several TV shows here in the US and I'm sure everywhere else that might be relevant and make a mission have a bit more meaning behind it.
I recently wasted tons of time checking out some of those airports and trips between them. It was especially gratifying seeing these RL areas in almost RL in FG.

Here is a short list of possibilities.

PAFA Fairbanks to PATA Tanana
PABE Bethel to PANI Aniak to PAUN Unalakleet to PAOM Nome or any combination.
PADQ Kodiak, PAKD Kodiak to PAUT Akutan or PASN St. Paul Island or PADK Adak Island or any combination.
Chitina CXC to PAMX McCarthy
PAEG Eagle to PACR Circle

or any of these to any of the surrounding areas.

**Comment by gilbertohasnofb** [comment]
@Juanvvc I am glad to hear you want to create a mission in South America! Here are some options of nice scenery there:

- For Chapada Diamantina region in Brazil, Lençóis Airport (SBLE) is a nice choice.

![image1](http://2.bp.blogspot.com/-OABg6RHjJ8U/VaWPYfR0YNI/AAAAAAAABFQ/fQ4er5SAjDU/s1600/fgfs-screen-016.png)

![image2](http://4.bp.blogspot.com/-NfYsYDcqjx4/VaWPZhXcnNI/AAAAAAAABFc/ZIAGx_mXha8/s1600/fgfs-screen-018.png)

- For the Amazon forest, close to Mount Roraima, Serra do Sol (SJMV) can be an option:

![image3](http://2.bp.blogspot.com/-OzCSdKBJPkE/Vb1FQAql1EI/AAAAAAAABJ0/j_yRRhYP7X8/s1600/fgfs-screen-013.png)

![image4](http://1.bp.blogspot.com/-agWtJrnn5l0/Vb1FQTtugwI/AAAAAAAABKA/bLKgC2Q_j-c/s1600/fgfs-screen-014.png)

![image5](http://4.bp.blogspot.com/-WXyGCG9VFtk/Vb1FZMnOm4I/AAAAAAAABLw/s0kRIbPJ_ew/s1600/fgfs-screen-033.png)

- But my favourite textures from that project are in the cerrado region in Brazil, for instance Pouso Alegre (SNZA):

![image6](http://2.bp.blogspot.com/-aDc4J2L_FZM/VYGy8PUONKI/AAAAAAAAA3o/hnPVyGnVEk0/s1600/fgfs-screen-003.png)

![image7](http://2.bp.blogspot.com/-qMKgfWzE8AY/VYG0WwOZ1ZI/AAAAAAAAA5k/mD3V-ADL_jI/s1600/fgfs-screen-007.png)

**Comment by onox** [comment]
You always make very nice screenshots @gilbertohasnofb :smile: You should add 5th to the forum/wiki (c172p page)

**Comment by gilbertohasnofb** [comment]
Thanks @onox :smile: I will do that, as well as make some small upgrade to our wiki page, there is a lot of old information there.

**Comment by gilbertohasnofb** [comment]
@onox I made some small upgrade to our wiki, see: http://wiki.flightgear.org/Cessna_172P

**Comment by wkitty42** [comment]
#5 would also make a good splash screen :wink:

**Comment by gilbertohasnofb** [comment]
> number 5 would also make a good splash screen

Done! I will create a PR for that :smile: 

@onox I am having a little trouble creating a new branch for this splash screen as to create a PR. First thing I do is a `git branch` to confirm I am on `master`, then I do:

```
$ git pull
Already up-to-date.
$ git checkout -b "new-splash"
M	Models/Effects/pontoon/pontoon.xml
Switched to a new branch 'new-splash'
```

Why is the file `Models/Effects/pontoon/pontoon.xml` being modified when I create the new branch? I thought that if I am on `master` and I do a `git checkout -b <new-branch-name>`, the new branch will start as an identical copy of master. 

Can you spot what I am doing wrongly?

**Comment by onox** [comment]
Can you do `git status`?

**Comment by gilbertohasnofb** [comment]
Oh, that indeed shows something strange, it shows a modification to that very file, which is really strange since I never edited any `pontoon.xml` myself. How can I simply ignore this modification and force my master branch to match the remote one?

**Comment by onox** [comment]
To make the `pontoon.xml` file return to its original state, you can do: `git checkout -- Models/Effects/pontoon/pontoon.xml`

**Comment by gilbertohasnofb** [comment]
Sure, that makes the files match, but `git status` still shows that that file has been modified:

```
$ git status
On branch master
Your branch is up-to-date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   Models/Effects/pontoon/pontoon.xml

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by onox** [comment]
You mean you don't want to throw your changed `pontoon.xml` away?

**Comment by gilbertohasnofb** [comment]
No, what I mean is that if my file is matching the remote one, there shouldn't be a single file to be modified when I do `git status`, right?

**Comment by onox** [comment]
Yes, but apparently you changed it, unless Git has become self-aware :wink:

**Comment by gilbertohasnofb** [comment]
hehehe of course, somehow I did it, I just can't even think of when/how I did that! Anyway, here is what is confusing me: I thought that by doing `git checkout -- Models/Effects/pontoon/pontoon.xml` I would force my local file to match the remote one, and therefore there would be no difference between the masters, thus nothing to commit, thus nothing being shown in `git status`. Makes sense?

**Comment by onox** [comment]
Btw, `git status` already explains how to stage changes (make them ready for commit) or discard them:

```
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)
```

So just create the new branch and stage the files you want. As long as you don't stage `Models/Effects/pontoon/pontoon.xml`, it won't be committed.

**Comment by gilbertohasnofb** [comment]
Ok, thanks a lot onox!

**Comment by Juanvvc** [comment]
Another option to be sure you have the same file than the repository is removing the file in your local filesystem and checking it out again.

```
$ rm Models/Effects/pontoon/pontoon.xml
$ git checkout Models/Effects/pontoon/pontoon.xml
```

**Comment by gilbertohasnofb** [comment]
I ended up cloning the whole repository again to make sure everything was fine, as that `git status` message was bugging me and I wanted to make sure all is correct (I have a better connection nowadays so it wasn't as painful as the old times). So I will open the PR for the splash screen any time soon.

**Comment by thevirtualfer** [comment]
It is occurring here to.

And i am sure i never edited pontoon.xml

I done @Juanvvc solution.
